### ISSUE TITLE
puppet_helper.groovy throwing error.. No such property: cred for class: Actions

### ISSUE THREAD

**Issue by albac** [jenkinsci/puppet-jenkins#624]
When creating credentials:

/usr/bin/java -jar /usr/lib/jenkins/jenkins-cli.jar -s http://127.0.0.1:8080 groovy /usr/lib/jenkins/puppet_helper.groovy create_or_update_credentials git-myjenkins-key '' '' 'Managed by Puppet' '/var/lib/jenkins/.ssh/id_rsa'

ERROR: Unexpected exception occurred while performing groovy command.
groovy.lang.MissingPropertyException: No such property: cred for class: Actions

**Comment by jhoblitt** [comment]
Is this from current master?

**Comment by albac** [comment]
Yes it is, I just took out the jenkins::credentials call and everything else works fine.

**Comment by jackdpeterson** [comment]
+1 on this issue.

**Comment by jhoblitt** [comment]
Hopefully, this is resolved via #628

**Comment by jackdpeterson** [comment]
I just checked master (forcing using master rather than tagged release in librarian-puppet) and I can confirm that the issue is solved.

Below is the stack trace for others.


```==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: ERROR: Unexpected exception occurred while performing groovy command.
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: groovy.lang.MissingPropertyException: No such property: cred for class: Actions
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:53)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.setGroovyObjectProperty(ScriptBytecodeAdapter.java:534)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at Actions.create_or_update_credentials(RemoteClass:396)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at java.lang.reflect.Method.invoke(Method.java:498)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1215)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1024)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at org.codehaus.groovy.runtime.InvokerHelper.invokePogoMethod(InvokerHelper.java:923)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethod(InvokerHelper.java:906)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodN(ScriptBytecodeAdapter.java:168)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at RemoteClass.run(RemoteClass:994)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at groovy.lang.GroovyShell.runScriptOrMainOrTestOrRunnable(GroovyShell.java:263)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at groovy.lang.GroovyShell.run(GroovyShell.java:518)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at groovy.lang.GroovyShell.run(GroovyShell.java:497)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at hudson.cli.GroovyCommand.run(GroovyCommand.java:86)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at hudson.cli.CLICommand.main(CLICommand.java:257)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at hudson.cli.CliManagerImpl.main(CliManagerImpl.java:92)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at java.lang.reflect.Method.invoke(Method.java:498)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at hudson.remoting.RemoteInvocationHandler$RPCRequest.perform(RemoteInvocationHandler.java:884)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at hudson.remoting.RemoteInvocationHandler$RPCRequest.call(RemoteInvocationHandler.java:859)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at hudson.remoting.RemoteInvocationHandler$RPCRequest.call(RemoteInvocationHandler.java:818)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at hudson.remoting.UserRequest.perform(UserRequest.java:153)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at hudson.remoting.UserRequest.perform(UserRequest.java:50)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at hudson.remoting.Request$2.run(Request.java:332)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at hudson.cli.CliManagerImpl$1.call(CliManagerImpl.java:63)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at hudson.remoting.CallableDecoratorAdapter.call(CallableDecoratorAdapter.java:18)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at hudson.remoting.CallableDecoratorList$1.call(CallableDecoratorList.java:21)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at jenkins.util.ContextResettingExecutorService$2.call(ContextResettingExecutorService.java:46)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
==> default: Notice: /Stage[main]/Main/Jenkins::Credentials[web]/Jenkins::Cli::Exec[create-jenkins-credentials-web]/Exec[create-jenkins-credentials-web]/returns: 	at java.lang.Thread.run(Thread.java:745)```
